packages/paste-core/components/avatar/__tests__/avatar.test.tsx

```diff
@@ -91,6 +101,7 @@ describe('Avatar', () => {
         <>
           <Avatar size="sizeIcon10" name="Simon Taggart" />
           <Avatar size="sizeIcon10" name="avatar example" src="/avatars/avatar2.png" />
+          <Avatar size="sizeIcon10" name="Simon Taggart" icon={UserIcon} />
```

Comment:
It might be nice to have a test in here that asserts the DOM structure you are creating.

As your implementation is opinionated on _how_ an icon should be treated, you should really write a test to make sure that opinion is maintained. 

Given and Icon and a name, avatar must set the icon to not be decorative and must set the title of the icon to the name. So you need to make sure that always stays true

Reply:
I like this, will do!

Reply:
@SiTaggart do you know of an existing example of testing the DOM structure somewhere else in the repo that we could take a look at?

Reply:
Not _exactly_ but disclosure does some DOM checking of expected attributes being set. 

You'd be doing similar but more DOM element expectations.

I'd expect to see a test to:

- Make sure there is an sag element
- Mak sure the SVG element _do not_ have aria-hidden="true"
- Make sure there is a <title> element in the SVG who's text content is equal to the name